### PR TITLE
chore(refactor): refactor Wasm engine and Sat construction

### PR DESCRIPTION
**Pull request by cohix**
This PR introduces the `engine2` package, which is a significant simplification of the previous `engine` package. A future PR will delete the `engine` package, and it was not deleted here for diff simplicity.

The construction of a `Sat` instance was also simplified.

### CODE REVIEW

File: sat/engine2/api/secrets.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package api
+
+import (
+	"github.com/pkg/errors"
+	"github.com/suborbital/e2core/sat/engine2/runtime/instance"
+)
+
+func (d *defaultAPI) GetSecretValueHandler() HostFn {
```
Comment (ospencer) [inline sat/engine2/api/secrets.go:8]: This should be implemented via WASI environment variables instead of a custom host function.

**Review by ospencer — COMMENTED** [review]
Overall looks reasonable, but I don't want the secrets host function in here.

**Comment by ospencer** [comment]
(Not that you added it, but no sense in bringing it over in the refactor.)

File: sat/engine2/api/secrets.go (hunk 1 above)
Comment (cohix) [reply sat/engine2/api/secrets.go]: Done

**Review by ospencer — APPROVED** [review]

**Pull request merged**